File: server/app/schemas/stats.py
from datetime import date as date_type
from pydantic import BaseModel


class OverviewStats(BaseModel):
    total_items: int
    active_items: int
    total_assets_value: float
    avg_daily_cost: float
    total_additional_costs: float


class CategoryStat(BaseModel):
    category_id: int | None
    category_name: str
    item_count: int
    total_value: float
    avg_daily_cost: float
    color: str | None


class DailyCostRankItem(BaseModel):
    id: int
    name: str
    daily_cost: float
    purchase_price: float
    usage_days: int
    status: str


class MonthlyTrendPoint(BaseModel):
    month: str
    spending: float
    item_count: int


class WarrantyAlert(BaseModel):
    id: int
    name: str
    warranty_expiry: date_type
    days_remaining: int
    purchase_price: float


class RecentItem(BaseModel):
    id: int
    name: str
    status: str
    purchase_price: float
    daily_cost: float
    created_at: str

File: web/src/api/stats.ts
import { api } from './client'

export interface OverviewStats {
  total_items: number
  active_items: number
  total_assets_value: number
  avg_daily_cost: number
  total_additional_costs: number
}

export interface CategoryStat {
  category_id: number | null
  category_name: string
  item_count: number
  total_value: number
  avg_daily_cost: number
  color: string | null
}

export interface DailyCostRankItem {
  id: number
  name: string
  daily_cost: number
  purchase_price: number
  usage_days: number
  status: string
}

export interface MonthlyTrendPoint {
  month: string
  spending: number
  item_count: number
}

export interface WarrantyAlert {
  id: number
  name: string
  warranty_expiry: string
  days_remaining: number
  purchase_price: number
}

export interface RecentItem {
  id: number
  name: string
  status: string
  purchase_price: number
  daily_cost: number
  created_at: string
}

export function getOverview() {
  return api<OverviewStats>('/stats/overview')
}

export function getByCategory() {
  return api<CategoryStat[]>('/stats/by-category')
}

export function getDailyCostRank(limit = 10, status?: string) {
  return api<DailyCostRankItem[]>('/stats/daily-cost-rank', { params: { limit, status } })
}

export function getTrends(period = 'month') {
  return api<MonthlyTrendPoint[]>('/stats/trends', { params: { period } })
}

export function getWarrantyAlerts(days = 30) {
  return api<WarrantyAlert[]>('/stats/warranty-alerts', { params: { days } })
}

export function getRecentItems(limit = 5) {
  return api<RecentItem[]>('/stats/recent', { params: { limit } })
}
